### PR TITLE
fix: [CI-9773]: Ignore test report upload failure for Run/RunTests step

### PR DESCRIPTION
**Pull request by rutvijmehta-harness**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/54235)

### CODE REVIEW

**Comment by rutvijmehta-harness** [comment]
trigger uts

**Comment by rutvijmehta-harness** [comment]
trigger uts

**Comment by rutvijmehta-harness** [comment]
trigger uts

**Comment by rutvijmehta-harness** [comment]
trigger messagecheck

**Comment by rutvijmehta-harness** [comment]
trigger ut5

File: product/ci/engine/grpc/tiproxy_handler.go [hunk 1]
```diff
@@ -200,36 +203,46 @@ func (h *tiProxyHandler) DownloadLink(ctx context.Context, req *pb.DownloadLinkR
...
 	if err != nil {
-		return nil, "", errors.Wrap(err, "failed to encode callgraph")
+		return nil, "", false, errors.Wrap(err, "failed to encode callgraph")
 	}
-	return encCg, msg, nil
+	return encCg, msg, false, nil
+}
+
+func isCgEmpty(cg *cgp.Callgraph) bool {
+	if len(cg.Nodes) == 0 && len(cg.TestRelations) == 0 && len(cg.VisRelations) == 0 {
```
Comment (hemanthmantri) [inline product/ci/engine/grpc/tiproxy_handler.go:242]: Shouldn't this be || condition?

**Review by hemanthmantri — APPROVED** [review]

File: product/ci/engine/grpc/tiproxy_handler.go (hunk 1 above)
Comment (rutvijmehta-harness) [reply product/ci/engine/grpc/tiproxy_handler.go:242]: No. Only if all are empty, we can conclude the CG to be empty.

Comment (hemanthmantri) [reply product/ci/engine/grpc/tiproxy_handler.go:242]: If a subset of these are empty, there's some error in the CG creation, right?

Comment (rutvijmehta-harness) [reply product/ci/engine/grpc/tiproxy_handler.go:242]: Depends on the agent. If there is data to upload, the client should upload regardless.

**Review by jyotiraval — APPROVED** [review]

**Pull request merged**